**Key Performance Indicators (KPIs): Data Analysis based on Vaccinen-Data**

In [8]:
import pandas as pd
import seaborn as sns
import plotly.io as pio
import plotly_express as px
import matplotlib.pyplot as plt 


#The file path along with the variable.
file_path = "Data/Folkhalsomyndigheten_Covid19_Vaccine.xlsx"
work_sheet = "Vaccinerade kommun och ålder"

#Reads the file into a dataframe
df = pd.read_excel(file_path, sheet_name= work_sheet)
df.head()

Län        Län_namn  Kommun     Kommun_namn  Ålder  Befolkning  \
0    1  Stockholms län     114  Upplands Väsby  12-15        2422   
1    1  Stockholms län     114  Upplands Väsby  16-17        1203   
2    1  Stockholms län     114  Upplands Väsby  18-29        6692   
3    1  Stockholms län     114  Upplands Väsby  30-39        7332   
4    1  Stockholms län     114  Upplands Väsby  40-49        6946   

   Antal minst 1 dos  Antal minst 2 doser  Antal 3 doser  Antal 4 doser  \
0               1206                 1046            NaN            NaN   
1                839                  755            NaN            NaN   
2               4887                 4469         1959.0            NaN   
3               5542                 5240         2878.0            NaN   
4               5592                 5429         3719.0            NaN   

   Andel minst 1 dos  Andel minst 2 doser  Andel 3 doser  Andel 4 doser  
0           0.497936             0.431874            NaN            NaN  
1           0.697423             0.627598            NaN            NaN  
2           0.730275             0.667812       0.292738            NaN  
3           0.755865             0.714675       0.392526            NaN  
4           0.805068             0.781601       0.535416            NaN

**- C) KPI: What is the overall percentage of the population in Sweden that has received the COVID-19 vaccin?**

 I'm analyzing COVID-19 vaccination data for various regions and age groups in Sweden. I begin by specifying the file path and sheet name for an Excel file containing the vaccination dataset. Using Pandas, I load the data into a DataFrame and display the total population of Sweden.

Next, I calculate and print vaccination statistics, including the percentage of the population with at least 1, 2, and 3 doses, along with the corresponding total number of vaccinated individuals. The script concludes by summarizing the total number of individuals with at least 1 dose, calculating the percentage of Sweden's population that has received this dose, and printing these key vaccination metrics.

Based on this dataset, the code provides a comprehensive overview of the COVID-19 vaccination status in Sweden. It offers insights into population statistics and detailed vaccination percentages for different dose categories, providing a detailed analysis of the vaccination landscape.

Data analysis: Loading data file and starting to process the data

In [9]:
"""How many are there in Sweden according to this dataset?"""

population = df["Befolkning"].sum()
print(f"The number of population in the dataset is: {population:,}")

The number of population in the dataset is: 9,092,790


In [10]:
"""Vaccinated as a percentage of Sweden's population with 1 dose"""

df["Antal minst 1 dos"] = df["Antal minst 1 dos"].fillna(0)
total_minst_1_dos = df["Antal minst 1 dos"].sum()
andel_minst_1_dos = (total_minst_1_dos / population) * 100 
print(f"Percentage of Sweden's population with at least 1 dose: {andel_minst_1_dos:.2f}%")
print("Total number of individuals with at least 1 dose in Sweden:", "{:,.0f}".format(total_minst_1_dos))


Percentage of Sweden's population with at least 1 dose: 85.90%
Total number of individuals with at least 1 dose in Sweden: 7,810,376


In [11]:
"""Vaccinated as a percentage of Sweden's population with 2 dose"""

df["Antal minst 2 doser"] = df["Antal minst 2 doser"].fillna(0)
total_minst_2_doser = df["Antal minst 2 doser"].sum()
andel_minst_2_doser = (total_minst_2_doser / population) * 100 
print(f"Percentage of Sweden's population with at least 2 doses: {andel_minst_2_doser:.2f}%")
print("Total number of individuals with at least 2 doses in Sweden:", "{:,.0f}".format(total_minst_2_doser))


Percentage of Sweden's population with at least 2 doses: 83.89%
Total number of individuals with at least 2 doses in Sweden: 7,627,585


In [12]:
"""Vaccinated as a percentage of Sweden's population with 3 dose"""

df["Antal 3 doser"] = df["Antal 3 doser"].fillna(0)
total_3_doser = df["Antal 3 doser"].sum()
andel_3_doser = (total_3_doser / population) * 100 
print(f"Percentage of Sweden's population with 3 doses: {andel_3_doser:.2f}%")
print("Total number of individuals with at least 3 doses in Sweden:", "{:,.0f}".format(total_3_doser))

Percentage of Sweden's population with 3 doses: 60.83%
Total number of individuals with at least 3 doses in Sweden: 5,531,250


In [13]:
"""Calculate the total number of vaccinated"""

antal_1_dos = df['Antal minst 1 dos'].sum()
totalt = antal_1_dos
andel_av_befolkningen = (totalt / population) * 100
print(f"Total number of vaccinated individuals: {totalt:,} st")
print(f"Percentage of Sweden's population vaccinated: {andel_av_befolkningen:.2f}%")


Total number of vaccinated individuals: 7,810,376 st
Percentage of Sweden's population vaccinated: 85.90%


In [14]:
"""Create a DataFrame for use with Plotly Express and generate a pie chart for:
 "Andel vaccinerade i Sverige" """

# Creating a DataFrame for use with Plotly Express
data = {
    "Dos": ["Minst 1 dos", "Ej vaccinerade"],
    "Antal": [totalt, population - totalt]}

df_plot = pd.DataFrame(data)

# Generating a pie chart using Plotly Express
fig = px.pie(
    df_plot,
    names="Dos",
    values="Antal",
    title="Andel vaccinerade i Sverige",
    labels={"Antal": "Number of Individuals"},
    hole=0.4,  
    color_discrete_sequence=["purple", "red"],  
    template="plotly_dark"  )

# Saving the chart as an HTML file
fig.write_html("Visualiseringar/3C.Andel vaccinerade kontra icke vaccinerade i Sverige.html")

# Displaying the plot
fig.show()

**Conclusion** 
Based on this dataset: many people have received at least one COVID-19 vaccine dose, 7,810,376 individuals in total. This corresponds to a high vaccination rate of 85.90% among Sweden's population. These numbers show significant efforts to lessen the virus's impact, boost community immunity, and contribute globally to the fight against the pandemic.